Task Liverpool/src/main/java/pages/SearchResultsPage.java

```diff
@@ -0,0 +1,152 @@
+        return Integer.parseInt(getElementText(numberOfResultsFromUpperRightLabel));
+    }
+
+    //Logic for clicking on the Results Page Right Arrow
+    public boolean clickNextPage(){
+        if (isElementDisplayed(rightArrow)){
+            String productId = getElementAttribute(findElement(searchResultsDisplayed), "data-prodid");
+            firstResult = By.cssSelector("li.m-product__card[data-prodid=\"" + productId + "\"]");
+            scrollToElement(rightArrow);
+            click(rightArrow);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public int getTotalNumberOfResultsDisplayed() {
+        int totalResults = findElements(searchResultsDisplayed).size();
+
+        while (clickNextPage()){
+            totalResults += findElementsAfterACurrentElementDisappear(searchResultsDisplayed, firstResult).size();
+        }
+        return totalResults;
+    }
+
+    public void clickListViewButton(){
+        click(listViewButton);
+    }
+
+    public void clickSortProductsBy (SortType sortType) throws InterruptedException {
+
+        click(sortByDropDownList);
+        switch (sortType) {
+            case RELEVANCE:
+                click(sortByRelevance);
+                break;
+
+            case LOWPRICE:
+                click(sortByLowPrice);
+                break;
+
+            case HIGHPRICE:
+                click(sortByHighPrice);
+                break;
+
+            case RATING:
+                click(sortByRating);
+                break;
+
+            case VIEWED:
+                click(sortByViewed);
+                break;
+
+            case SOLD:
+                click(sortBySold);
+                break;
+        }
+        Thread.sleep(1000);
+    }
+
+    public void getPricesInt(){
+        pricesList = findElements(productPrice);
+        pricesIntList = new ArrayList<>();
+
+        for (WebElement prices: pricesList) {
+            pricesIntList.add(Integer.parseInt(prices.getText().substring(1).replace(",", "")));
+        }
+    }
+
+    public void addValuesToPricesIntListFromMoreResultsPages(){
+        while (clickNextPage()){
+            pricesList = findElementsAfterACurrentElementDisappear(productPrice, firstResult);
+            for (WebElement prices: pricesList) {
+                pricesIntList.add(Integer.parseInt(prices.getText().substring(1).replace(",", "")));
+            }
+        }
+    }
+
+    public boolean checkPriceIsSorted(){
+        getPricesInt();
+        addValuesToPricesIntListFromMoreResultsPages();
+
+        for (int i = 0; i < pricesIntList.size() - 1; i++){
+            boolean compare = (pricesIntList.get(i) >= pricesIntList.get(i + 1));
+            if (!compare){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public void clickFilterLowerThan$500() throws InterruptedException {
+        findElement(filterLowerThan$500Button).click();
+        Thread.sleep(1000);
```

Comment:
do not use Thread.sleep

Reply:
I created a new method in BasePage called findElementsAfterTextChanges where I wait until the text of the label for the numberOfResultsFromUpperRightLabel locator has a different value. I had to make this because this text was the only thing I could find being updated at the same time in the DOM as the products in the screen after selecting how to Filter them.